pkg/apis/wasm/v1alpha1/wasm_module_validation.go

```
@@ -32,24 +32,87 @@ func (as *WasmModule) Validate(ctx context.Context) *apis.FieldError {
-	if ass.Source.Image == "" {
+	imageFieldPath := "image"
+	// A WasmModule must specify its image. The 'image' field
+	// points to the OCI image containing the Wasm module.
+	if ass.Image == "" {
 		//goland:noinspection GoDfaNilDereference
 		errs = errs.Also(apis.ErrMissingField(imageFieldPath))
 	}
 
 	// TODO: validate remote registry for accessibility of parsed ref
-	if _, err := reference.Parse(ass.Source.Image); err != nil {
+	if _, err := reference.Parse(ass.Image); err != nil {
```

golangci-lint (2.5.0)

[error] avoid inline error handling using `if err := ...; err != nil`; use plain assignment `err := ...`

(noinlineerr)